### PR TITLE
JavaScript: object destructuring with parameter decorators

### LINKED ISSUE #6394: Prettier breaks lines when arg has decorator and is destructured
**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgCpyYwBiEATgLbowwCWUA5jsADpQ44ACAsgK4zVa0ABRk4MXmSi4AvAD4cAZQEAzFQEo2HLAE8oYHOgoIAJsvRrhWjlwCCZBlZD0ADvxYhN7G8By0TADQ4AHShRqbmagAi1Og4AL5IOLbGUGaqKgCSUG4w1upJAApkEBS0mHAAPJEqCqzeHGISUjhEFioA3NbxbD1QbGAY2PiEJORUNPRM9Rw8-IIiTZLSOPJKGV7amHoG4Wk1Vg12Dk6u7p7W2qiYEAAy0AwAcuHc6DoARnAAanBkOveMADyZAAogA3BAAhi-JIpCIZbK5fJFEplCrVDJ1S44JYtNpqLrePrxEABEAQFx0aCYZCgdBkEoAd0K9IQNJQ6DBEH8pJA7zI6DAAGtxIoXIKpsgYGReHAyQALGAUVAAdXltHgmHFYDgijZGtoYI1OmQ4GwvPoFTIMGK6AYVGQKnQ1zlIAAVpgAB4AIQFwtF4Vu9DgjudFTJHs9iimqDgAEVeBB4KGXWTxWQrab3uh3jpUNBeS4yPQYCr-DB5cgABwABjTJQqKoFLlNRcIvwhvIAjon4MUKeyQFgALRQOBwEwT3liHu0MS2+3oFPhkAVMpSmWuzAx+O9kNIJ2pkACd5lkwV5AAJjJ0vQtFQUwAwqUHSgoNAQ2TeBU8Dn2YeKnieIgA)

**Input:**
```ts
class TestFormatting {
  @Mutation(returns => Staff)
  async amendStaff(
    @Arg("input")
    { id, ...amendStaffData }: AmendStaffInput
  ): Promise<Staff> {
    return staff;
  }
}
```

**Output:**
```ts
class TestFormatting {
  @Mutation(returns => Staff)
  async amendStaff(@Arg("input")
  {
    id,
    ...amendStaffData
  }: AmendStaffInput): Promise<Staff> {
    return staff;
  }
}
```

**Expected behavior:**
Keep the destructured variables in one line (keep original formatting).
```ts
class TestFormatting {
  @Mutation(returns => Staff)
  async amendStaff(
    @Arg("input")
    { id, ...amendStaffData }: AmendStaffInput
  ): Promise<Staff> {
    return staff;
  }
}
```

This is the behavior for normal parameters that are not destructured:

```ts
class TestFormatting {
  @Mutation(returns => Staff)
  async amendStaff(
    @Arg("input")
    alsoLongNameMaybeVeryLongOrEvenLonger: AmendStaffInput
  ): Promise<Staff> {
    return staff;
  }
}
```

### PR DESCRIPTION
**Pull request by sosukesuzuki**
Fix #6394 

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
somebody in @prettier/core can review? if approved, I'll fix conflicts.

File: CHANGELOG.unreleased.md [hunk 1]
```diff
@@ -415,6 +415,37 @@ const foo = [abc, def, ghi, jkl, mno, pqr, stu, vwx, yz] as (
 )[];
 ```
 
+#### JavaScript: Fix ugly formatting on object destructuring with parameter decorators ([#6411] by [@sosukesuzuki])
+
+Previously, destructured object under parameter decorators breaks.
```
Comment (lydell) [inline CHANGELOG.unreleased.md]:
```suggestion
Previously, Prettier formatted decorators for destructured parameters in a weird way. Now, parameter decorators are placed just above the parameter they belong to.
```

**Comment by sosukesuzuki** [comment]
@lydell Thanks for your review. I updated CHANGELOG.unreleased.md and fixed conflicts.

**Pull request merged**